nemo/collections/tts/waveglow_modules.py

```diff
@@ -177,20 +181,28 @@ def __init__(
         wn_kernel_size: int = 3,
         sigma: float = 0.6,
     ):
+        NonTrainableNM.__init__(self)  # For NeuralModule API
         self._sigma = sigma
-        # self.sample_rate = sample_rate  # Done in parent class
-        super().__init__(
-            sample_rate=sample_rate,
-            n_mel_channels=n_mel_channels,
-            n_flows=n_flows,
-            n_group=n_group,
-            n_early_every=n_early_every,
-            n_early_size=n_early_size,
-            n_wn_layers=n_wn_layers,
-            n_wn_channels=n_wn_channels,
-            wn_kernel_size=wn_kernel_size,
-        )
+        self.sample_rate = sample_rate  # Done in parent class
         self._removed_weight_norm = False
+        wavenet_config = {
+            "n_layers": n_wn_layers,
+            "n_channels": n_wn_channels,
+            "kernel_size": wn_kernel_size,
+        }
+        self.waveglow = (
+            waveglow
+            if waveglow
+            else WaveGlow(
+                n_mel_channels=n_mel_channels,
+                n_flows=n_flows,
+                n_group=n_group,
+                n_early_every=n_early_every,
+                n_early_size=n_early_size,
+                WN_config=wavenet_config,
+            )
+        )
```

Comment:
Why is WaveGlowInferNM now a wrapper for the WaveGlow nn.module?

Reply:
because we do
`            return self.waveglow.forward(mel_spectrogram, z, sigma=self._sigma)
`